server/src/main/java/io/crate/planner/operators/InsertFromValues.java

```diff
@@ -268,9 +268,8 @@ public void execute(DependencyCarrier dependencies,
         var shardedRequests = new ShardedRequests<>(builder::newRequest, RamAccounting.NO_ACCOUNTING);
 
         for (Row row : rows) {
-            Item item = grouper.apply(shardedRequests, row, true);
-
             try {
+                Item item = grouper.apply(shardedRequests, row, true);
                 // Primary Key and CLUSTERED BY check is already done in grouper -> RowShardResolver, both cannot be null.
```

Comment:
Is it a bug (and should it be backported)?

Tests are passing but I feel like it can cause some troubles as code flow became different. 
Before https://github.com/crate/crate/pull/14364 is was fine to call `grouper.apply` before try block as exceptions used to be swallowed.

Then grouper was slightly refactored and now I think we need to move grouper.apply under try as well. 

g`rouper.apply` in `executeBulk` is already under try.

Reply:
> Is it a bug (and should it be backported)?

You tell me :)

It looks like the grouper can at least throw a CircuitBreakingException, which would get swallowed without this change.
So looks like a bug to me and should be backported with a release notes entry.

Reply:
Thanks for the `CircuitBreakingException` pointer. 

>which would get swallowed without this change. 

Looks like other way around?

Even before https://github.com/crate/crate/pull/14364 and on current master it's not swallowed in grouper and also not under `try` in `InsertFromValues`, ie was not finished by `consumer.accept(null, t);`

I will investigate a bit more the consequence of not failing the row properly/not consuming exception.

Reply:
> Looks like other way around?

With swallowed I mean that we don't propagate the error to the client as response, but instead it bubbles up into the netty thread (which afaik triggers the `exceptionCaught` handler. In the PostgresWireProtocol that would just log an error)

Reply:
Looks like it actually never throws because of a `NO_ACCOUNTING` circuit breaker:

https://github.com/crate/crate/blob/2ad0da463a32a65ec2b43763979da51a5860813b/server/src/main/java/io/crate/planner/operators/InsertFromValues.java#L268

So as far as I can tell there is no user facing issue.